### PR TITLE
Kraken: Fix redundant stop schedules when stop points impacted

### PR DESCRIPTION
**Pull request by xlqian**
http://jira.canaltp.fr/browse/NAVITIAII-2144

### CODE REVIEW

File: source/routing/get_stop_times.cpp [hunk 1]
```diff
@@ -141,15 +141,19 @@ get_stop_times(const std::vector<routing::JppIdx>& journey_pattern_points,
...
                    const type::VehicleProperties& vehicle_properties) {
 
     std::set<const type::MetaVehicleJourney*> meta_vjs;
     auto insert_meta_vj = [&](const nt::VehicleJourney& vj ) {
         assert(vj.meta_vj);
         meta_vjs.insert(vj.meta_vj);
     };
-    for (const auto* vj: jp.discrete_vjs) { insert_meta_vj(*vj); }
+    for (const auto* vj: jp.discrete_vjs) {
+        if(vj->isBaseScheduled()) {
```
Comment (pbougue) [inline source/routing/get_stop_times.cpp]: You have to check more than that, no?
Not sure, but we have to know the freshness we want, and provide the corresponding one, I'd say (defaulting to base).
The departure board uses this if I'm correct.

Comment (stifoon) [inline source/routing/get_stop_times.cpp]: In a calendar mode, we don't want other freshness than base_shedule. We don't want real_time for a "days of scholar week".

Comment (xlqian) [inline source/routing/get_stop_times.cpp]: we care only about the base scheduled VJs when `calendar_id` is provided

File: source/type/type.h [hunk 2]
```diff
@@ -510,7 +510,9 @@ struct VehicleJourney: public Header, Nameable, hasVehicleProperties {
     int16_t utc_to_local_offset() const; //in seconds
 
     RTLevel realtime_level = RTLevel::Base;
-
+    bool isBaseScheduled() const{
```
Comment (kinnou02) [inline source/type/type.h]: should be in snake case

Comment (xlqian) [inline source/type/type.h]: oops  :scream:

**Comment by TeXitoi** [comment]
OK for me.

File: source/routing/get_stop_times.cpp (hunk 1 above)
Comment (pbougue) [inline source/routing/get_stop_times.cpp]: as orally debated, a rename of get_all_stop_times and get_stop_times mentioning calendar would be nice

File: source/type/type.h (hunk 2 above)
Comment (pbougue) [inline source/type/type.h]: and I would remove the 'd': is_base_schedule

**Comment by pbougue** [comment]
ok for me

**Pull request merged**